[WIP] Sep18 MeasurementConfiguration

rhomspuron opened the pull request.

rhomspuron committed:
Add MGConfiguration class

rhomspuron committed:
Adapt to use MGConfiguration

rhomspuron committed:
Merge zreszela/sep18 branch

rhomspuron committed:
Adapt PoolAcquisition

rhomspuron committed:
Add method to prepare acquisition variable

Include variables for the acquisition actions.
Implement methods to prepare the configuration for the acquisition
actions.

rhomspuron committed:
Adapt PoolAcquisition to use MGConfiguration API

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

rhomspuron committed:
Allow to create a raw configuration

rhomspuron committed:
Add internal getters to PoolAcquisitionBase

Add getters to extract the controllers, timer and monitor according to
the acquisition synchronization (hardware or software).

rhomspuron committed:
Use the main_element configuration

rhomspuron committed:
Adapt helpers to use MGConfiguration class

rhomspuron committed:
Adapt aquisition test to use MGConfiguration class

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

rhomspuron committed:
Remove typo error

Remove line copied by mistake

rhomspuron committed:
Rename MGConfiguration to MeasurementConfiguration

The class will be used by the measurement groups and the counter/timer
channels.

rhomspuron committed:
Rename internal variable

Change "mg" to "parent" as it is more appropriate to the new use of
the class.

rhomspuron committed:
Fix bug on test_aquisition

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

rhomspuron committed:
Rename internal methods

rhomspuron committed:
Add trigger gate configuration

Add method to extract the trigger/gate configurations

rhomspuron committed:
Use trigger/gate configuration from MeasurementConfiguration

rhomspuron committed:
Adapt to use MeasurementConfiguration class

rhomspuron committed:
Adapt test to use MeasurementConfiguration class

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

rhomspuron committed:
Remove unused properties

rhomspuron committed:
Add new attributes for acquisition actions

Add lists for timers and monitor for each the acquisition action type.

rhomspuron committed:
Use MeasurementConfiguration API

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -241,14 +139,17 @@ def event_received(self, *args, **kwargs):
+    def prepare(self, config, repetitions):
         """Prepare measurement."""
-        timers = []  # config.get_timers(enabled=True)
+        timers = config.sw_sync_timers_enabled + \
```
IMHO all the timers should be accessible directly from the configuration i.e. the API client should not make sums of the lists if he just wants all of them.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -241,14 +139,17 @@ def event_received(self, *args, **kwargs):
                 self._0d_acq.stop_action()
 
-    def prepare(self, config, nr_of_starts):
+    def prepare(self, config, repetitions):
```
This name should be maintained `nr_of_starts`

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -241,14 +139,17 @@ def event_received(self, *args, **kwargs):
             axis = timer.axis
             timer_ctrl = timer.controller
             ctrl = timer_ctrl.ctrl
-            ctrl.PrepareOne(axis, nr_of_starts)
```
This name should be maintained `nr_of_starts`

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -271,44 +172,38 @@ def run(self, *args, **kwargs):
+            if len(config.ctrl_0d_sync):
                 zerod_acq_kwargs = dict(kwargs)
-                zerod_acq_kwargs['config'] = zerod_acq_cfg
+                # TODO: Ask why
```
Please exlain in detail the question.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -481,24 +399,27 @@ def start_action(self, *args, **kwargs):
 
+        if integ_time is not None and mon_count is not None:
+            raise RuntimeError('The acquisition must have only one role: '
+                               'timer or count')
```
It should be "timer or monitor"

rhomspuron replied:
ok

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -654,6 +575,31 @@ class PoolAcquisitionHardware(PoolAcquisitionBase):
 
+    def _get_ctrls(self):
+        """
+        Method to get the controller dict for the acquisition type
```
I would remove this docsting (it is explained in the super class) or add information that in this action it gets the controllers which are hardware synchronized.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -654,6 +575,31 @@ class PoolAcquisitionHardware(PoolAcquisitionBase):
+
+    def _get_timer(self):
+        """
+        Method to get the controller dict for the acquisition type
```
I would remove this docsting (it is explained in the super class) or add information that in this action it gets the controllers which are hardware synchronized.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -654,6 +575,31 @@ class PoolAcquisitionHardware(PoolAcquisitionBase):
+        """
+        return self.main_element.configuration.hw_sync_timer
+
+    def _get_monitor(self):
```
I would remove this docsting (it is explained in the super class) or add information that in this action it gets the controllers which are hardware synchronized.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -724,6 +670,31 @@ def __init__(self, main_element, name="AcquisitionSoftware", slaves=None):
 
+    def _get_ctrls(self):
+        """
+        Method to get the controller dict for the acquisition type
```
I would remove this docsting (it is explained in the super class) or add information that in this action it gets the controllers which are software synchronized.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -724,6 +670,31 @@ def __init__(self, main_element, name="AcquisitionSoftware", slaves=None):
+
+    def _get_timer(self):
+        """
+        Method to get the controller dict for the acquisition type
```
I would remove this docsting (it is explained in the super class) or add information that in this action it gets the controllers which are software synchronized.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -724,6 +670,31 @@ def __init__(self, main_element, name="AcquisitionSoftware", slaves=None):
+
+    def _get_monitor(self):
+        """
+        Method to get the controller dict for the acquisition type
```
I would remove this docsting (it is explained in the super class) or add information that in this action it gets the controllers which are software synchronized.

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -914,7 +885,7 @@ def start_action(self, *args, **kwargs):
             items = self.get_elements()
         cfg = kwargs['config']
```
In other acquisition actions you obtain configuration from the main_element. Why not here?

reszelaz commented on src/sardana/pool/poolacquisition.py:
```diff
@@ -914,7 +885,7 @@ def start_action(self, *args, **kwargs):
         cfg = kwargs['config']
 
-        pool_ctrls_dict = dict(cfg['controllers'])
+        pool_ctrls_dict = dict(cfg.controllers)
```
This should get the 0D controllers only.

reszelaz reviewed (changes requested):
I have not finished with the review. I just reviewed the poolacquisition module. I will continue tomorrow. Menwhile I submit this review.

You need to merge here the latest sep18 branch.

Then I noticed two tests failing
```
======================================================================
ERROR: test_generation (sardana.pool.poolcontrollers.test.test_DummyTriggerGateController.PoolDummyTriggerGateTestCase)
Testing generation(synchronization=[{<SynchParam.Delay: 0>: {<SynchDomain.Time: 0>: 0}, <SynchParam.Total: 1>: {<SynchDomain.Time: 0>: 0.02}, <SynchParam.Active: 2>: {<SynchDomain.Time: 0>: 0.01}, <SynchParam.Repeats: 3>: 10}])
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zreszela/workspace/taurus/lib/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/home/zreszela/workspace/sardana/src/sardana/pool/poolcontrollers/test/test_DummyTriggerGateController.py", line 60, in generation
    self.tg_action.start_action(*args, **kwargs)
  File "/home/zreszela/workspace/sardana/src/sardana/pool/poolsynchronization.py", line 144, in start_action
    cfg = self.main_element.configuration
AttributeError: 'PoolTriggerGate' object has no attribute 'configuration'

======================================================================
ERROR: test_generation_2 (sardana.pool.poolcontrollers.test.test_DummyTriggerGateController.PoolDummyTriggerGateTestCase)
Testing generation(synchronization=[{<SynchParam.Delay: 0>: {<SynchDomain.Time: 0>: 0}, <SynchParam.Total: 1>: {<SynchDomain.Time: 0>: 0.1}, <SynchParam.Active: 2>: {<SynchDomain.Time: 0>: 0.03}, <SynchParam.Repeats: 3>: 0}])
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zreszela/workspace/taurus/lib/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/home/zreszela/workspace/sardana/src/sardana/pool/poolcontrollers/test/test_DummyTriggerGateController.py", line 60, in generation
    self.tg_action.start_action(*args, **kwargs)
  File "/home/zreszela/workspace/sardana/src/sardana/pool/poolsynchronization.py", line 144, in start_action
    cfg = self.main_element.configuration
AttributeError: 'PoolTriggerGate' object has no attribute 'configuration'

----------------------------------------------------------------------
Ran 184 tests in 651.965s

FAILED (errors=2, skipped=7, expected failures=1)
```

rhomspuron committed:
Fix flake8 errors

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

rhomspuron committed:
Change message error

rhomspuron committed:
Fix docstrings

rhomspuron committed:
Adapt Pool0DAquisition to use MeasurementConfiguration API

rhomspuron committed:
Remove TODO

rhomspuron committed:
Use nr_of_starts instead of repetitions

rhomspuron committed:
Adapt test to use MeasurementConfiguration API

rhomspuron committed:
Remove unused property

rhomspuron committed:
Merge branch 'sep18' of github.com:reszelaz/sardana into sep18_config

rhomspuron committed:
Adapt load_configuration method

rhomspuron committed:
Fix bug on get_timer method

reszelaz commented on src/sardana/pool/poolmeasurementgroup.py:
```diff
@@ -123,100 +124,200 @@ def _to_fqdn(name, logger=None):
+    def __init__(self, parent=None):
+        self._parent = None
+        if parent is not None:
+            self._parent = weakref.ref(parent)()
```
I think tha this way you are creating a real reference and not a weakref.

rhomspuron replied:
Yes, you're right, I'll change for the proxy: https://docs.python.org/2.7/library/weakref.html

> weakref.proxy(object[, callback])
    Return a proxy to object which uses a weak reference. This supports use of the proxy in most contexts instead of requiring the explicit dereferencing used with weak reference objects. The returned object will have a type of either ProxyType or CallableProxyType, depending on whether object is callable. Proxy objects are not hashable regardless of the referent; this avoids a number of problems related to their fundamentally mutable nature, and prevent their use as dictionary keys. callback is the same as the parameter of the same name to the ref() function.

reszelaz commented on src/sardana/pool/poolmeasurementgroup.py:
```diff
@@ -123,100 +124,200 @@ def _to_fqdn(name, logger=None):
-        self._moveable_obj = None
-        # by default software synchronizer initial domain is set to Position
-        self._sw_synch_initial_domain = SynchDomain.Position
+        self.use_fqdn = True
```
I would keep it *private*.

rhomspuron replied:
ok

reszelaz commented on src/sardana/pool/poolmeasurementgroup.py:
```diff
@@ -123,100 +124,200 @@ def _to_fqdn(name, logger=None):
-        return PoolAcquisition(self, acq_name)
+    @configuration.setter
+    def configuration(self, config=None):
+        self._build_configuration(config)
```
Shouldn't it check if the config is None?

reszelaz commented on src/sardana/pool/poolmeasurementgroup.py:
```diff
@@ -267,84 +368,82 @@ def _build_channel_defaults(self, channel_data, channel):
         return channel_data
 
-    def _build_configuration(self):
+    def _build_configuration(self, config=None):
```
I don't like the idea to allow to pass here the config

rhomspuron replied:
This is not the user_configuration and the idea is to have only one method to prepare and validate the configuration. We can design another implementation

reszelaz reviewed (changes requested):
I reviewed the poolmeasurementgroup module. There are some small comments and a question about the `_build_configuration` API that we should discuss.

reszelaz commented:
Design of the `MeasurementConfiguration` API:

```
MeasurementConfiguration

get_configuration_for_user
set_configuration_from_user

OPTION 1: 
there are two method to take the structure {ctrls:[channels]}

get_controllers(action_type=None, enabled=True) => []

get_elements([ctrls], enabled=True) => {ctrl:[channels]}

get_timers(action_type=None, enabled=True) => []
    if action_type == "synchronization": raise ValueError()
    if action_type == "zerod": raise ValueError()

get_monitors(action_type=None, enabled=True) => []
    if action_type == "synchronization": raise ValueError()
    if action_type == "zerod": raise ValueError()

get_master(action_type, mode=None) = master
    if action_type == "synchronization" and mode is not None: raise ValueError()
    if action_type == "zerod": raise ValueError()


OPTION 2: 
there was not protection for disable timer/monitor of the controller

get_ctrls_channels(action_type=None, enabled=True) => {ctrl:[channels]}

get_timers(action_type=None, enabled=True) => {ctrl:timer}
    if action_type == "synchronization": raise ValueError()
    if action_type == "zerod": raise ValueError()

get_monitors(action_type=None, enabled=True) => {ctrl:monitor}
    if action_type == "synchronization": raise ValueError()
    if action_type == "zerod": raise ValueError()

get_master(action_type, mode=None) => master
    if action_type == "synchronization" and mode is not None: raise ValueError()
    if action_type == "zerod": raise ValueError()


OPTION 3: 
get the timer/monitor of the controller is not efficient and is not easy to read.

get_ctrls_channels(action_type=None, enabled=True) => {ctrl:[channels]}

get_<timers/monitor>(ctrls=None, enabled=True) => [timer/monitor]
    if ctrls=None and enabled = None -> all timer
    if ctrls=[xxx] and enabled = True/False if there is any controller with 
    channel in different state raise Exception
    if ctrls=[xxx] and enable = None do not check state of timer/monitor

get_master(action_type, mode=None) => master
    if action_type == "synchronization" and mode is not None: raise ValueError()
    if action_type == "zerod": raise ValueError()


OPTION 4:

get_master is not explicit and node argument is too complex and may not be coherent with
synchronization

get_ctrl_channels(action_type=None, enabled=True) => {ctrl:[channels]}

get_ctrl_<timer/monitor>(ctrls=None, enabled=True) => {ctrl: timer/monitor}
    if ctrls=None and enabled = None -> all timer
    if ctrls=[xxx] and enabled = True/False if there is any controller with 
    channel in different state raise Exception
    if ctrls=[xxx] and enable = None do not check state of timer/monitor

get_master(action_type, mode=None) => master
    if action_type == "synchronization" and mode is not None: raise ValueError()
    if action_type == "zerod": raise ValueError()

                     
                     
OPTION 5:

get_ctrl_channels(action_type=None, enabled=None) => {ctrl:[channels]}
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_ctrl_<timer/monitor>(ctrls=None, enabled=None) => {ctrl: timer/monitor}
    if ctrls=None and enabled = None -> all timer
    if ctrls=None and enabled = True/False -> all enabled/disabled timer/monitor
    if ctrls=[xxx] and enabled = True/False if there is any controller with 
        channel in different configuration raise RuntimeException
    if ctrls=[xxx] and enabled = None do not check state of timer/monitor

get_master_<timer/monitor>(action_type, enabled=None) => master
    if action_type in ["synchronization", "zerod"]: raise ValueError()
    if enabled = True/False: raise RuntimeError() if master timer/monitor is
        in different configuration
    if enabled = None do not check state of timer/monitor

TODO: get_master_synch() => master

OPTION 6:
* channels are wrapped Pool elements by Channel(PoolActionItem) class
* this options does not isolate concepts of configuration and acquisition action

get_timerable_ctrl_channels(acq_synch=None, enabled=None) => {ctrl:[channels]}
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_zerod_ctrl_channels(enabled=None) => {ctrl:[channels]}
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_synch_ctrl_channels(enabled=None) => {ctrl:[channels]}
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_ctrl_<timer/monitor>(ctrls=None) => {ctrl: timer/monitor}
    if ctrls=None -> all timer
    if ctrls=[xxx] -> not check state of timer/monitor

get_master_<timer/monitor>_software() => master
    do not check state of timer/monitor

get_master_<timer/monitor>_software_start() => master
    do not check state of timer/monitor

TODO: get_master_synch() => master

FINAL:

Specific classes with configuration:

class ConfigurationItem(object):
    
    def __init__(self, element, conf=None):
        self._element = weakref.ref(element)
        if conf:
            self.__dict__.update(conf)

    def __getattr__(self, name):
        return getattr(self.element, name)

    def get_element(self):
        """Returns the element associated with this item"""
        return self._element()

    def set_element(self, element):
        """Sets the element for this item"""
        self._element = weakref.ref(element)

    element = property(get_element)

class ControllerConfiguration(ConfigurationItem):
    """Configuration: 'timer', 'monitor', 'synchronization', 'channels'"""
    
class ChannelConfiguration(ConfigurationItem):
    """Configuration: 'id', 'enabled', 'output', 'plot_type', 'plot_axes',
                      'label', 'scale', 'plot_color'"""
    

get_timerable_ctrls(acq_synch=None, enabled=None) => [ControllerConfiguration]
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_zerod_ctrls(enabled=None) => [ControllerConfiguration]
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_synch_ctrls(enabled=None) => [ControllerConfiguration]
    enabled = True => only enabled
    enabled = False => only disabled
    enabled = None => all

get_master_<timer/monitor>_software() => ChannelConfiguration
    do not check state of timer/monitor

get_master_<timer/monitor>_software_start() => ChannelConfiguration
    do not check state of timer/monitor

TODO: get_master_synch() => master

```

reszelaz committed:
Refactors acqusition base action

reszelaz committed:
Adapt acquisition test case to new API

reszelaz committed:
Add TODO for measurement configuration

rhomspuron committed:
Remove unused code

rhomspuron committed:
Remove get configuration method

Remove internal method to get the configuration for each action type.

rhomspuron committed:
Add explicit parameters on start_action method

rhomspuron committed:
Add python 3 compatibility

rhomspuron committed:
Adapt to use PoolAcquisitionBase class

rhomspuron committed:
Rename start_action parameters

rhomspuron committed:
Update start_action docstring

rhomspuron committed:
Do use_fqdn to private attribute

rhomspuron committed:
Fix weak reference

Use a weak reference of the parent instead of the object itself.

reszelaz committed:
Refactor acquisition and its tests to use ControllerConfiguration objects

ControllerConfiguration objects will have information about its channels
and its master channel. Assume it in start_action and tests.

rhomspuron committed:
Add configuration helpers classes

Add classes to help the use of the static measurement configuration.

rhomspuron committed:
Add method to ConfigurationItems classes

rhomspuron committed:
Refactor configuration classes

Refactor classes to use them with the synchronizer controllers and
channels

rhomspuron committed:
Add SyncrhonizerConfiguration class

rhomspuron committed:
Add enabled attribute to ConfigurationItem

Add enabled attribute to the ConfigurationItem class with True as
default value.

rhomspuron committed:
Refactor MeasurementConfiguration class

Refactor class to update the internals attribute at set the
configuration.
Add protections to validate the configuration set by the user.

rhomspuron committed:
Change acq_synch lookup table

Use two lookup table one for channel and another for controller.
Use as key the element instead of the id.

rhomspuron committed:
Adapt classes to use new MeasurementConfiguration API

rhomspuron committed:
Change start_action API

Do explicit the parameters needed by the action

rhomspuron committed:
Adapt test to use new start_action API

rhomspuron committed:
Include the 0D channel on the test

rhomspuron committed:
Adapt test to new start_action API

rhomspuron committed:
Change 0D start_action API

rhomspuron committed:
Update SynchronizationDescription

Change integration_time attribute name to active_time
Add passive_time attribute

rhomspuron committed:
Update prepare method

reszelaz committed:
(minor) Remove unused imports

reszelaz committed:
Refactor acquisition test

reszelaz committed:
Add test for software start acquisition

reszelaz committed:
Implement action_loop of software start acquisition

reszelaz committed:
Fix test of software start acquisition

reszelaz committed:
Refactor acquisition tests to facilitate development of new tests

reszelaz committed:
Add hardware start acquisition test

reszelaz committed:
Move AttributeListener to util module

rhomspuron committed:
Fix minor error

rhomspuron committed:
Fix flake8 errors

rhomspuron committed:
Remove unused imports

rhomspuron committed:
Adapt synchronization test to start_action new API

rhomspuron committed:
Remove unused helper

rhomspuron committed:
Remove unused helper

reszelaz committed:
(minor) Refactor of measurement group tests

reszelaz committed:
Add test to software and hardware start acquisition

reszelaz committed:
Add software start acquisition action to super acquisition action

rhomspuron committed:
Add defaults values for measurement configuration keys

rhomspuron committed:
Remove not mandatory keys from the configuration helper

reszelaz committed:
Merge branch 'sep18_config' of https://github.com/rhomspuron/sardana into sep18_config

reszelaz committed:
Select master timer/monitor for software start acquisition

reszelaz committed:
Start refactoring of acquisition action

rhomspuron committed:
Allow the dynamic attribute creation on PoolActionItem

Update the wrapper attributes by using a configuration dict on the class
initialization.

Access to the element attributes from the wrapper.

rhomspuron committed:
Fix typo mistake

rhomspuron committed:
Add changed attribute to MeasurementConfiguration

rhomspuron committed:
Use PoolActionController instead of CtrlActionItem

rhomspuron committed:
Fix error on adding jobs

rhomspuron committed:
Move configuration loading to the action prepare

rhomspuron committed:
Adapt to use the action prepare

rhomspuron committed:
Remove unused method

rhomspuron committed:
Rename parameter

Change the parameter config by attrs that is more appropriate
according to its use.

rhomspuron committed:
Include AcqSynch and AcqSynchType

rhomspuron committed:
Add classes to __all__

rhomspuron committed:
Refactor PoolSynchronization

* Add synchronization as a mandatory start_action paramter.
* Remove unused code.
* Rename local variables to make them more redeable.

rhomspuron committed:
Add Acquisition wrappers

Add AcqConfigurationItem and AcqController classes to wrap the
configuration object and add necessary information for the acquisition
actions, such as the master channel used in the timerable controllers.

rhomspuron committed:
Add helpers to get the items for the actions

rhomspuron committed:
Remove unused import

rhomspuron committed:
Remove unsued Channel class

rhomspuron committed:
Rename local variables

rhomspuron committed:
Refactor PoolAcquistion run and prepare method

Move the preparation logic from the run to the prepare method.

rhomspuron committed:
Adapt test_acquisition

rhomspuron committed:
Adapt test_poolsynchronization

rhomspuron committed:
Adapt test_synchronization

rhomspuron committed:
Add classes to __all__ variable

reszelaz committed:
Remove Preparable interface and add PrepareOne to Loadable

rhomspuron committed:
Remove *args from start_action methods

rhomspuron committed:
Change local variable name

Use index instead of value to be more explicit.

rhomspuron committed:
Change start_action methods

Adapt each start_action method according to the action requirements.

reszelaz committed:
Return SoftrwareTrigger acq synch by default

reszelaz committed:
Consider software start acq action ctrls in top acq action

reszelaz committed:
Update MG's master timer and monitor in user configuration

reszelaz committed:
PEP8

reszelaz committed:
Fix FQDN convertion for unit test cases

reszelaz committed:
Fix several issues with MG configuration

* External attributes - Tango
* Validation whether ctrl's timer/monitor is enabled
* Creation of measurement group (fill default configuration)
* element assignment to actions based on synchronizer's type (sw vs. hw)

reszelaz committed:
Add infrastructure for 2D tests

reszelaz committed:
Refactor start of acquisition action

reszelaz committed:
Fix dummy trigger/gate controller tests

reszelaz committed:
Set full_name in dummy configurations

reszelaz committed:
Remove useless test

reszelaz committed:
Fix PEP8

reszelaz commented:
Seem like all tests are green again. Merging. Thanks a lot!
